bigtable-client-core-parent/bigtable-hbase/src/main/java/org/apache/hadoop/hbase/client/BigtableAsyncRegistry.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright 2017 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.hadoop.hbase.client;
+
+import com.google.api.core.InternalApi;
+import com.google.common.collect.ImmutableList;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import net.bytebuddy.ByteBuddy;
+import net.bytebuddy.dynamic.DynamicType;
+import net.bytebuddy.dynamic.loading.ClassLoadingStrategy;
+import net.bytebuddy.dynamic.loading.InjectionClassLoader;
+import net.bytebuddy.implementation.FixedValue;
+import net.bytebuddy.implementation.InvocationHandlerAdapter;
+import net.bytebuddy.implementation.MethodCall;
+import net.bytebuddy.matcher.ElementMatchers;
+import org.apache.hadoop.conf.Configuration;
+
+/**
+ * Bigtable implementation of org.apache.hadoop.hbase.client.ConnectionRegistry or
+ * org.apache.hadoop.hbase.client.AsyncRegistry depending on the HBase 2 version. The default HBase
+ * 2 implementation provided by ZKAsyncRegistry / ZKConnectionRegistry assumes a ZooKeeper
+ * environment, which is not the case for Bigtable.
+ *
+ * <p>This class is injected via the system property: "hbase.client.registry.impl". For further
+ * details See HConstants#CLIENT_CONNECTION_REGISTRY_IMPL_CONF_KEY,
+ * AsyncRegistry#createAsyncConnection() for hbase < 2.3 and
+ * ConnectionFactory#createAsyncConnection() for hbase >= 2.3.
+ *
+ * <p>For internal use only - public for technical reasons.
+ */
+@InternalApi("For internal usage only")
+public class BigtableAsyncRegistry {
```

Comment:
to confirm my understanding - this will work as a drop in replacement here? https://github.com/googleapis/java-bigtable-hbase/blob/main/bigtable-hbase-2.x-parent/bigtable-hbase-2.x-integration-tests/pom.xml#L38

Reply:
This is updated in this pr: https://github.com/googleapis/java-bigtable-hbase/pull/3452 , BigtableEnv will call BigtableConnection() to create the connection for integration tests, so we don't need to define `google.bigtable.registry.impl` anymore, and BigtableConnection will handle generating the class. I'll also clean up this property in the pom file.